ow_power_system/src/PowerSystemNode.cpp

```diff
@@ -263,30 +268,155 @@ void PowerSystemNode::injectFault (const string& fault_name,
+        custom_warning_displayed = true;
+      }
+    }
+    else
+    {
+      // Get user-entered file directory
+      if (! ros::param::getCached("/faults/custom_fault_profile", designated_file))
+      {
+        ROS_ERROR("Failed to fetch custom fault name.");
+        return;
+      }
+
+      // Append the current fault directory to the stored file path.
+      current_fault_directory = ros::package::getPath("ow_power_system") + "/profiles/" + designated_file;
+      
+      if (current_fault_directory != saved_fault_directory || end_fault_warning_displayed)
+      {
+        // It's a different fault profile from whatever the previous stored directory was, OR the
+        // previous fault ran its course, so reset index.      
+        // Load a new fault profile from the provided directory.
+        try
+        {
+          ifstream file(current_fault_directory);
+          if (current_fault_directory.substr(current_fault_directory.size() - 4) != ".csv")
+          {
+            throw -1;
+          }
+          else if (file.fail())
+          {
+            throw -2;
+          }
+          if (end_fault_warning_displayed && designated_file == saved_file)
+          {
+            ROS_INFO_STREAM("Reloading " << designated_file << "...");
+          }
+          else if (saved_fault_directory != "N/A")
+          {
+            ROS_INFO_STREAM("Loading " << designated_file << " and unloading " << saved_file << "...");
+          }
+          else
+          {
+            ROS_INFO_STREAM("Loading " << designated_file << "...");
+          }
+          loadCustomFaultPowerProfile(current_fault_directory);
+          custom_fault_ready = true;
+          end_fault_warning_displayed = false;
+          // TEMP: Uncomment this and the other 'TEMP' line code to re-enable restricting power faults:
+          // When uncommented, the user will only be able to use either high power draw or the custom
+          // fault feature, never both in a single simulation. Behavior of both at once is unpredictable.
+          //CUSTOM_FAULT_ACTIVATED = true;
+          saved_fault_directory = current_fault_directory;
+          saved_file = designated_file;
+          index = 0;
+
+          ROS_WARN_STREAM_ONCE("Custom power faults may exhibit unexpected results. Caution is advised.");
+          //ROS_WARN_STREAM_ONCE("Note that custom power faults cannot be combined with high power draw.");
+          ROS_INFO_STREAM(saved_file.substr(0, saved_file.size() - 4) << " activated!");
+        }
+        catch (int err_val)
+        {
+          switch(err_val)
+          {
+            case -1:
+              ROS_WARN_STREAM("Invalid file name entered; Deactivate fault and try again.");
+              break;
+            case -2:
+              ROS_WARN_STREAM("Could not find CSV in the 'profiles' directory with the provided name. Deactivate fault and try again.");
```

Comment:
Would be helpful to include the filename in these messages.  If this is hard or impractical, don't worry about it.